File: 0912 sort-an-array/solution.java
class Solution {
    public int[] sortArray(int[] nums) {
        int max = Integer.MIN_VALUE,min = Integer.MAX_VALUE;
        for(int i:nums) {
            max=Math.max(max,i);
            min = Math.min(min,i);
        }
        int offset = -min;
        int[] arr = new int[max - min + 1];
        for(int i:nums) arr[i + offset]++;
        int[] ans = new int[nums.length];
        int k =0;
        for(int i=0;i<arr.length;i++) {
            while(arr[i] > 0 ) {
                ans[k++] = i - offset;
                arr[i]--;
            }  
        }
        return ans;
    }
}

File: 0912 sort-an-array/solution.cpp
class Solution {
public:

    void divide(vector<int>& nums,int srt,int end) {
        if(srt >= end) return;
        int mid = srt+(end-srt)/2;
        divide(nums,srt,mid);
        divide(nums,mid+1,end);
        conquer(nums,srt,end,mid);
    }
    void conquer(vector<int>& nums,int srt,int end,int mid) {
        vector<int> arr(end - srt + 1);
        int i=srt,j=mid+1,k=0;
        while(i<=mid && j<=end) {
            if(nums[i]<=nums[j]) arr[k++] = nums[i++];
            else arr[k++] = nums[j++];
        }
        while(i<=mid) arr[k++] = nums[i++];
        while(j<=end) arr[k++] = nums[j++];

        for(i=0,j=srt;i<arr.size();i++,j++) {
            nums[j] = arr[i];
        }
    }
    vector<int> sortArray(vector<int>& nums) {
       divide(nums,0,nums.size()-1);
       return nums;
    }
};
